cli/module_generate.go

```diff
@@ -0,0 +1,726 @@
+		if !d.IsDir() && strings.HasPrefix(d.Name(), templatePrefix) {
+			destPath := filepath.Join(module.ModuleName, strings.ReplaceAll(path, templatePrefix, ""))
+			debugf(c.App.Writer, c.Bool(debugFlag), "\tRendering file %s", destPath)
+
+			tFile, err := templates.Open(filepath.Join(languagePath, path))
+			if err != nil {
+				return err
+			}
+			defer utils.UncheckedErrorFunc(tFile.Close)
+			tBytes, err := io.ReadAll(tFile)
+			if err != nil {
+				return err
+			}
+
+			tmpl, err := template.New(path).Parse(string(tBytes))
+			if err != nil {
+				return err
+			}
+
+			//nolint:gosec
+			destFile, err := os.Create(destPath)
+			if err != nil {
+				return err
+			}
+			defer utils.UncheckedErrorFunc(destFile.Close)
+
+			err = tmpl.Execute(destFile, module)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Generate stubs for the resource.
+func generateStubs(c *cli.Context, module moduleInputs) error {
+	debugf(c.App.Writer, c.Bool(debugFlag), "Generating %s stubs", module.Language)
+	switch module.Language {
+	case python:
+		return generatePythonStubs(module)
+	default:
+		return errors.Errorf("cannot generate stubs for language %s", module.Language)
+	}
+}
+
+func generatePythonStubs(module moduleInputs) error {
+	venvName := ".venv"
+	cmd := exec.Command("python3", "--version")
+	_, err := cmd.Output()
+	if err != nil {
+		return errors.Wrap(err, "cannot generate python stubs -- python runtime not found")
+	}
+	cmd = exec.Command("python3", "-m", "venv", venvName)
+	_, err = cmd.Output()
+	if err != nil {
+		return errors.Wrap(err, "cannot generate python stubs -- unable to create python virtual environment")
+	}
+	defer utils.UncheckedErrorFunc(func() error { return os.RemoveAll(venvName) })
+
+	script, err := scripts.ReadFile(filepath.Join(scriptsPath, "generate_stubs.py"))
+	if err != nil {
+		return errors.Wrap(err, "cannot generate python stubs -- unable to open generator script")
+	}
+	//nolint:gosec
+	cmd = exec.Command(filepath.Join(venvName, "bin", "python3"), "-c", string(script), module.ResourceType,
+		module.ResourceSubtype, module.Namespace, module.ModuleName, module.ModelName)
+	out, err := cmd.Output()
+	if err != nil {
+		return errors.Wrap(err, "cannot generate python stubs -- generator script encountered an error")
+	}
+
+	mainPath := filepath.Join(module.ModuleName, "src", "main.py")
+	//nolint:gosec
+	mainFile, err := os.Create(mainPath)
+	if err != nil {
+		return errors.Wrap(err, "cannot generate python stubs -- unable to open file")
+	}
+	defer utils.UncheckedErrorFunc(mainFile.Close)
+	_, err = mainFile.Write(out)
+	if err != nil {
+		return errors.Wrap(err, "cannot generate python stubs -- unable to write to file")
+	}
+
+	return nil
+}
+
+func getLatestSDKTag(c *cli.Context, language string) (string, error) {
+	var repo string
+	if language == python {
+		repo = "viam-python-sdk"
+	}
```

Comment:
(minor, opt) probably good for this to be a switch statement for future extensibility.